website/src/components/SearchPage/SearchFullUI.spec.tsx

```diff
@@ -234,7 +234,10 @@ describe('SearchFullUI', () => {
         await userEvent.click(field1Checkbox);
         const closeButton = await screen.findByRole('button', { name: 'Close' });
         await userEvent.click(closeButton);
-        await waitForElementToBeRemoved(() => screen.queryByText('Toggle the visibility of search fields'));
+        const toggleText = screen.queryByText('Toggle the visibility of search fields');
+        if (toggleText) {
```

Comment:
Under which conditions would we not expect the toggleText to be there? Why does it need to be guarded?

Reply:
Hmm. The thing didn't work anymore after switching to happy-dom, I'm thinking it might be a timing issue. Given that this is only done to make sure the pop-up is gone (if I understand correctly) I thought it would be fine to do it this way. I didn't investigate further into the reasons

Reply:
no the await is still there as it is in front of `waitForElementToBeRemoved`. If you modify the strings you can actually check that the test still works.

The new code simply first checks if the element is there before waiting for it to be removed.

Reply:
> The element(s) given to waitForElementToBeRemoved are already removed. waitForElementToBeRemoved requires that the element(s) exist(s) before waiting for removal.